play-services-location/core/src/main/kotlin/org/microg/gms/location/manager/LocationManager.kt

```diff
@@ -47,7 +47,12 @@ class LocationManager(private val context: Context, override val lifecycle: Life
     private val postProcessor by lazy { LocationPostProcessor() }
     private val lastLocationCapsule by lazy { LastLocationCapsule(context) }
     val database by lazy { LocationAppsDatabase(context) }
-    private val requestManager by lazy { LocationRequestManager(context, lifecycle, postProcessor, database) { onRequestManagerUpdated() } }
+    private val requestManager by lazy { LocationRequestManager(context, lifecycle, postProcessor, database) {
+        lifecycleScope.launchWhenStarted {
+            ensurePermissions()
```

Comment:
This doesn't seem to be the right spot for this.

Reply:
Do you have any good suggestions, since I haven't found a more suitable place.